proto/decentraland/common/rect.proto

```diff
@@ -0,0 +1,9 @@
+﻿syntax = "proto3";
+package decentraland.common;
+
+message Rect {
+  float x = 1;
+  float y = 2;
+  float width = 3;
+  float height = 4; 
+}
```

Comment:
by convention rects are:
```suggestion
message Rect {
  float top = 1;
  float left = 2;
  float right = 3;
  float bottom = 4; 
}
```

If we want to leverage the We may find another name for this struct instead? Like Rectangle? This rect won't work for `texture_slices` for example.

Reply:
After running some mental exercises I've come to the conclusion that using the "good ol" Rect with left top right and bottom is the best for this case. Here is the exercise:


assuming:
```
  Rect texture_slices = 4; // default=(1/3, 1/3, 1/3, 1/3)
  Rect uv = 5;
```

### texture_slices

Texture slices uses (by implicit convention) the distance to the borders. Which cannot be achieved by specifying width and height in uniform space.

The default `(1/3, 1/3, 1/3, 1/3)` specifies the distance to the borders. 

This is familiar in unity as seen in the following image, and it is also the convention for CSS and Godot's [NinePatchRect](https://docs.godotengine.org/en/3.0/classes/class_ninepatchrect.html) ![](https://docs.unity3d.com/uploads/Main/9SliceSprites-3.jpg)

### uv

### Conclusion 1

Uvs must also be a rect. Since those are uniform space coordinates, the implementations of all UVs is bound to the absolute texture coordinate of the vertexes. If we were to use `x+width` instead of `right` then, to calculate the texture coordinate of `right` the engine should sum both `x+width`.

This is not a real deal-breaker, but for the sake of consistency, I'd use the rect as bottom right instead of x+width and y+height.

### Conclusion 1, revisited

After some research, I've found that in Unity is almost trivial to implement custom UVs in the UI using the class [UnityEngine.UIElements.MeshGenerationContext](https://docs.unity3d.com/Packages/com.unity.ui@1.0/api/UnityEngine.UIElements.MeshGenerationContext.html). Given this fact, I'd recommend to rollback the uvs to the same mechanism used for the in-world plane, an array of 8 floats. To enable more flexibility and to maintain consistency in the SDK

Reply:
If we stick with
```
message Rect {
  float top = 1;
  float left = 2;
  float right = 3;
  float bottom = 4; 
}
```
for defining borders it gives a different meaning to the following convention:

![image](https://user-images.githubusercontent.com/118179774/206715301-6b250b4b-77e7-466a-afa7-b1bb666f592e.png)

If in the rest of the framework we use `Rect` as a distance from the top-left corner, for slices it is a distance from the respective edge.
It is not a deal breaker but it may be confusing as we will be expanding the functionality

Reply:
>After some research, I've found that in Unity is almost trivial to implement custom UVs in the UI using the class [UnityEngine.UIElements.MeshGenerationContext](https://docs.unity3d.com/Packages/com.unity.ui@1.0/api/UnityEngine.UIElements.MeshGenerationContext.html). Given this fact, I'd recommend to rollback the uvs to the same mechanism used for the in-world plane, an array of 8 floats. To enable more flexibility and to maintain consistency in the SDK

I would not go this way as if we introduce `Rotation` for `Transform`, this functionality will be redundant/confusing.
I would make a decision regarding `Transform`s first (in the future iteration).

Reply:
Regardless of introducing rotation. We are already doing the UV mapping in 3D entities. I'd say that let's introduce the UVs to leverage all the texture atlas libraries and helpers we have in place for it.